ovos_date_parser/dates_ca.py

```
@@ -13,6 +14,92 @@
+    pattern = r"(?P<value>\d+(?:\.?\d+)?)(?:\s+|\-){unit}[s]?"
+
+    text = text.replace("í", "i").replace("é", "e").replace("ñ", "n").replace("mesos", "mes")
+    text = numbers_to_digits_ca(text)
+
+    for (unit_en, unit_ca) in time_units.items():
+        unit_pattern = pattern.format(
+            unit=unit_ca[:-1])  # remove 's' from unit
+        time_units[unit_en] = 0
+
+        def repl(match):
+            time_units[unit_en] += float(match.group(1))
```

Ruff

Function definition does not bind loop variable `unit_en`

(B023)